addons/delivery/models/stock_picking.py

```diff
@@ -95,13 +95,20 @@ def _compute_bulk_weight(self):
             ['picking_id', 'product_id', 'product_uom_id', 'qty_done'],
             lazy=False, orderby='qty_done asc'
         )
+        products_by_id = {
+            product_res['id']: (product_res['uom_id'][0], product_res['weight'])
+            for product_res in
+            self.env['product.product'].with_context(active_test=False).search_read(
+                [('id', 'in', list(set(grp["product_id"][0] for grp in res_groups)))], ['uom_id', 'weight'])
+        }
         for res_group in res_groups:
-            product_id = self.env['product.product'].browse(res_group['product_id'][0])
+            uom_id, weight = products_by_id[res_group['product_id'][0]]
+            uom = self.env['uom.uom'].browse(uom_id)
             product_uom_id = self.env['uom.uom'].browse(res_group['product_uom_id'][0])
             picking_weights[res_group['picking_id'][0]] += (
                 res_group['__count']
-                * product_uom_id._compute_quantity(res_group['qty_done'], product_id.uom_id)
-                * product_id.weight
+                * product_uom_id._compute_quantity(res_group['qty_done'], uom)
```

Comment:
don't you have the same issue with uom breaking the prefetch at each iteration ?